### PR TITLE
[overlay] Silence warnings

### PR DESCRIPTION
**Pull request by JaSpa**
These changes silence the lot of warnings appearing when building the overlays. They were mostly about the by now explicit handling required for optionals.
In the case of `URLResourceValues` I eliminated the other “old” stuff like roundtrips for correct number-bridging as well.

cc @parkera

### CODE REVIEW

**Comment by parkera** [comment]
I'm a little nervous about relying on the behavior of _SwiftValue's bridging logic here. I guess now we assume that it will always convert things like optional to `NSNull`, but I don't know if that is a contractual promise or not.

I can't recall now why I added the contains method on the `URLResourceValues` struct but I guess we don't use it so it's fine to remove.

File: stdlib/public/SDK/SpriteKit/SpriteKitQuickLooks.swift.gyb [hunk 1]
```diff
@@ -18,22 +18,14 @@ extension ${Self} : CustomPlaygroundQuickLookable {
...
     // we could send a Raw, but I don't want to make a copy of the
     // bytes for no good reason make an NSImage out of them and
     // send that
 #if os(OSX)
-    if let data = data {
-      return .sprite(NSImage(data: data as Data))
-    } else {
-      return .sprite(NSImage())
-    }
+    return .sprite(data.flatMap(NSImage.init(data:)) ?? NSImage())
```
Comment (jrose-apple) [inline stdlib/public/SDK/SpriteKit/SpriteKitQuickLooks.swift.gyb]: This seems overlay dense to me. Let's just leave the if-let.

**Comment by parkera** [comment]
That would help. I specifically wanted NSNull here because that's the contract of the ObjC API (and I missed it on the first pass; this code was a bug fix).

**Comment by parkera** [comment]
Also, we should double check the one that turns `[String]` into `NSObject`. I can't recall which key needs that but we should make sure it still works.

**Comment by parkera** [comment]
At least we have a test for the quarantine one.

**Comment by JaSpa** [comment]
The only key with a type of `[String]` seems to be `.tagNamesKey` for which there is no setter.

**Comment by JaSpa** [comment]
So, could someone please run the tests/merge? I don't have write access

**Comment by parkera** [comment]
@swift-ci please test

**Comment by moiseev** [comment]
@swift-ci Please smoke test Linux platform

**Comment by moiseev** [comment]
Argh... I meant:
@swift-ci Please smoke test macOS platform

**Comment by JaSpa** [comment]
Ping?

**Comment by CodaFi** [comment]
#6506 supersedes this patch so I'm going to close it.

**Pull request closed without merging**